### ISSUE TITLE
bug: Node accepts connections when configured for just discovery

### ISSUE THREAD

**Issue by alrevuelta** [waku-org/nwaku#2892]
### Problem

A node that is configured as a "boostrap" node without mounting any protocol such as `relay`, `store`, etc is meant to act as a "discovery only bootstrap peer". See example configuration:

```
--relay=false\
--discv5-discovery=true\
--discv5-enr-auto-update=True\
--pubsub-topic=/waku/2/rs/66/0\
--cluster-id=66
```

However, with this configuration, this node is still mounting the libp2p switch + protocol such as `metadata` or `identify` and accept external connections. This doesn't make much sense since the node is just providing discovery capabilities on discv5 level and no service is offered on libp2p level.

In other words, a node not mounting any waku protocol, shouldn't accept any connection. The proposed solution would be to not mount the libp2p switch if `relay`, `store`, ... are all false.

This behaviour can be reproduced with [waku-simulator](https://github.com/waku-org/waku-simulator) where the [boostrap node](https://github.com/waku-org/waku-simulator/blob/master/run_bootstrap.sh) acts as discovery-only.

![image](https://github.com/waku-org/nwaku/assets/8811422/ad166adf-b28e-41ff-bf58-181d4a61c43d)

@AlbertoSoutullo 
@stubbsta

**Comment by fryorcraken** [comment]
The problem statement doesn't clarify why one would want to run a bootstrap node with discv5 only.
